src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -248,3 +250,109 @@ def _try_fetch(self) -> Iterable[Mount]:
             f"SELECT * FROM {escape_sql_identifier(self._schema)}.{escape_sql_identifier(self._table)}"
         ):
             yield Mount(*row)
+
+
+@dataclass
+class TableInMount:
+    format: str
+    is_partitioned: bool
+
+
+class TablesInMounts(CrawlerBase[Table]):
+
+    def __init__(
+        self,
+        backend: SqlBackend,
+        ws: WorkspaceClient,
+        inventory_database: str,
+        mc: Mounts,
+        include_mounts: list[str] | None = None,
+        filter_paths_in_mount: list[str] | None = None,
+    ):
+        super().__init__(backend, "hive_metastore", inventory_database, "tables", Table)
+        self._dbutils = ws.dbutils
+        self._mc = mc
+        self._include_mounts = include_mounts
+
+        irrelevant_patterns = {'_SUCCESS', '_committed_', '_started_'}
+        if filter_paths_in_mount:
+            irrelevant_patterns.update(filter_paths_in_mount)
+        self._fiter_paths = irrelevant_patterns
+
+    def snapshot(self) -> list[Table]:
+        """
+        Takes a snapshot of tables in the specified catalog and database.
+
+        Returns:
+            list[Table]: A list of Table objects representing the snapshot of tables.
+        """
+        return self._snapshot(partial(self._try_load), partial(self._crawl))
+
+    def _try_load(self) -> Iterable[Table]:
+        """Tries to load table information from the database or throws TABLE_OR_VIEW_NOT_FOUND error"""
+        for row in self._fetch(f"SELECT * FROM {escape_sql_identifier(self.full_name)}"):
+            yield Table(*row)
+
+    def _crawl(self):
+        all_mounts = self._mc.snapshot()
+        all_tables = []
+        for mount in all_mounts:
+            if self._include_mounts and mount.name not in self._include_mounts:
```

Comment:
First check in "if" condition is redundant, as self._include_mounts is already defined and checking key in set always result to amortized complexity of O(1).  So, just checking key absence is more than enough. Also, if possible to check there isn 't duplicate values in snapshots, then we can modify to get set of mounts instead of list of mounts.

Reply:
self._include_mounts is None by default, hence  mount.name not in self._include_mounts can fail.

There cannot be duplicate mount values as Databricks workspaces cannot have mounts with the same name